napari/_qt/menus/file_menu.py

```diff
@@ -200,19 +44,18 @@ def _rebuild_samples_menu(self):
         ):
             multiprovider = len(samples) > 1
             if multiprovider:
-                menu = self.open_sample_menu.addMenu(plugin_name)
+                sub_menu_id = f'napari/file/samples/{plugin_name}'
+                sub_menu = (
+                    MenuId.SAMPLES,
+                    SubmenuItem(
+                        submenu=sub_menu_id, title=trans._(plugin_name)
+                    ),
+                )
+                SUBMENUS.append(sub_menu)
             else:
-                menu = self.open_sample_menu
+                sub_menu_id = MenuId.SAMPLES
 
             for samp_name, samp_dict in samples.items():
-                display_name = samp_dict['display_name'].replace("&", "&&")
-                if multiprovider:
-                    action = QAction(display_name, parent=self)
-                else:
-                    full_name = menu_item_template.format(
-                        plugin_name, display_name
-                    )
-                    action = QAction(full_name, parent=self)
 
                 def _add_sample(*args, plg=plugin_name, smp=samp_name):
```

Comment:
Any idea why `*args` is needed here? It does not seem to be used.
The dependency injection for
```python
            def _add_sample(
                qt_viewer: QtViewer,
                plg=plugin_name,
                smp=samp_name,
            ):
```

in `_build_samples_menu`, does not work with `*args`. Works fine for napari samples without `*args`

Any ideas @DragaDoncila @nclack ?

Reply:
I don't think we need to deal with this file at all, Isn't the entire file_menu.py to be dropped at this point?

Reply:
pinging @brisvag & @Czaki , due to discussion I saw [here](https://github.com/napari/napari/pull/5513/files#r1090445759) : would either of you be able to explain my question [above](https://github.com/napari/napari/pull/4865/files#r1001484592)? Also is `*args` and `*_` equivalent? How do they differ? Thank you